File: tests/testdata/codegen/single-or-none-pp/nodejs/single-or-none.ts
import * as khulnasoft from "@khulnasoft/khulnasoft";

function singleOrNone<T>(elements: khulnasoft.Input<T>[]): khulnasoft.Input<T> {
    if (elements.length != 1) {
        throw new Error("singleOrNone expected input list to have a single element");
    }
    return elements[0];
}

export const result = singleOrNone([1]);

File: tests/testdata/codegen/single-or-none-pp/go/single-or-none.go
package main

import (
	"fmt"

	"github.com/khulnasoft/khulnasoft/sdk/v3/go/khulnasoft"
)

func singleOrNone[T any](elements []T) T {
	if len(elements) != 1 {
		panic(fmt.Errorf("singleOrNone expected input slice to have a single element"))
	}
	return elements[0]
}

func main() {
	khulnasoft.Run(func(ctx *khulnasoft.Context) error {
		ctx.Export("result", khulnasoft.Float64(singleOrNone([]float64{
			1,
		})))
		return nil
	})
}
